staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -521,10 +545,137 @@ func (a *Authenticator) AuthenticateToken(ctx context.Context, token string) (*a
 		}
 	}
 
+	var claimsUnstructured *unstructured.Unstructured
+	// Convert the claims to unstructured so that we can evaluate the CEL expressions
+	// against the claims. This is done once here so that we don't have to convert
+	// the claims to unstructured multiple times in the CEL mapper for each mapping.
+	// Only perform this conversion if any of the mapping or validation rules contain
+	// CEL expressions.
+	// TODO(aramase): In the future when we look into making distributed claims work,
+	// we should see if we can skip this function and use a dynamic type resolver for
+	// both json.RawMessage and the distributed claim fetching.
+	if a.celMapper.Username != nil || a.celMapper.Groups != nil || a.celMapper.UID != nil || a.celMapper.Extra != nil || a.celMapper.ClaimValidationRules != nil {
+		if claimsUnstructured, err = convertObjectToUnstructured(&c); err != nil {
+			return nil, false, fmt.Errorf("oidc: could not convert claims to unstructured: %w", err)
+		}
+	}
```

Comment:
should feature enablement be checked here? there's a _lot_ of code movement / change in AuthenticateToken and ValidateAuthenticationConfiguration that makes it hard to tell if the existing code path for non-CEL configurations is intact

I was hoping it would be more obvious that existing code paths were intact if the feature gate was disabled or no CEL capabilities were used in the config, either via a pattern like:
```
if feature-and-use-of-cel-condition {
  ... new code
} else {
  ... old code
}
```

or a factoring that:
1. takes each step being done in token validation / mapping and turns it into an interface
2. has a non-cel implementation for each interface that uses the existing code
3. has a cel implementation for each interface
4. constructs the appropriate implementation based on whether cel was used

Reply:
> should feature enablement be checked here?

we are checking for feature enablement in `validation.go`. `New` method that constructs the authenticator gets the compiled CELMapper for validation and if the feature is not enabled, the subfields are nil.

> I was hoping it would be more obvious that existing code paths were intact if the feature gate was disabled or no CEL capabilities were used in the config, either via a pattern like

We are following this pattern here.
1. If expressions are set and feature gate isn't enabled, we get an error from `CompileAndValidateJWTAuthenticator` and fail to create an Authenticator.
2. If expressions are set and feature gate is enabled, we have mappers for the subfields which contain expressions.
   - I've moved the username and groups computation code to their own functions, `getUsername` and `getGroups` respectively.  Within those functions, we have the old code as-is and the new code is inside an if loop if the subfield mapper isn't `nil`.
   - `getUID`, `getExtra` is net new code that didn't exist before.
   - `claimValidationRules` is an array and can contain a mix of the old (`.claim` and `.requiredValue`) and new (`.expression` and `.message`). The old code is kept as-is and we have added check with cel expressions if the subfield is not nil.
   - `userValidationRules` is net new, so that has been added now.

There is a lot of `if feature-enabled` checks in the validation code to guard against:
1. using expressions in username, groups claim mapping, claim validation when feature gate not set
2. using UID, Extra claim mapping when feature gate not set 
3. using user validation rules when feature gate not set